packages/ui/src/providers/datamapper.provider.test.tsx

```diff
@@ -44,4 +44,33 @@ describe('DataMapperProvider', () => {
     expect(prevTree! !== nextTree!).toBeTruthy();
     expect(prevTree!.children[0] === nextTree!.children[0]).toBeTruthy();
   });
+
+  it('resetMappingTree should reset the mappings', async () => {
+    let initialized = false;
+    let reset = false;
+    let tree: MappingTree;
+    const TestRefreshMappingTree = () => {
+      const { mappingTree, refreshMappingTree, resetMappingTree } = useDataMapper();
+      useEffect(() => {
+        if (!initialized) {
+          mappingTree.children.push(new FieldItem(mappingTree, {} as IField));
+          refreshMappingTree();
+          initialized = true;
+        } else if (!reset) {
+          resetMappingTree();
+          reset = true;
+        } else {
+          tree = mappingTree;
+        }
+      }, [mappingTree, refreshMappingTree, resetMappingTree]);
+      return <div data-testid="testdiv"></div>;
+    };
+    render(
+      <DataMapperProvider>
+        <TestRefreshMappingTree></TestRefreshMappingTree>
+      </DataMapperProvider>,
+    );
+    await waitFor(() => tree);
+    expect(tree!.children.length).toEqual(0);
```

Comment:
Just as a side comment, have you had the chance to see the `renderHook` utility?

```
renderHook(() => useDataMapper, { wrapper: YOUR_PROVIDER_GOES_HERE });
```

The advantage is that you can manipulate the output of the hook directly, without relying on the `useEffect` hook or creating a temporary component for it.